src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/MacRazorProjectHostBase.cs

```diff
@@ -272,13 +274,22 @@ private static string ResolveFallbackIntermediateOutputPath(IMSBuildEvaluatedPro
             }
 
             var projectDirectory = projectProperties.GetValue(MSBuildProjectDirectoryPropertyName);
-            var joinedPath = Path.Combine(projectDirectory, intermediateOutputPathValue);
+            var normalizedProjectDirectory = ToPlatformFilePath(projectDirectory);
+            var joinedPath = Path.Combine(normalizedProjectDirectory, intermediateOutputPathValue);
             if (!Directory.Exists(joinedPath))
             {
                 return null;
             }
 
             return joinedPath;
         }
+
+        /// <summary>
+        /// Project system file paths get returned in windows based foramts. Meaning they typically have `\` as their path separator. Because of this in order to
+        /// interoperate with local file system APIs (all of the Path / Directory etc. APIs) we need to convert the paths to use `/` as the path separator.
+        /// </summary>
+        /// <param name="filePath">A project system based file path.</param>
+        /// <returns>A file path that can be used with File system APIs</returns>
+        private static string ToPlatformFilePath(string filePath) => filePath.Replace('\\', '/');
```

Comment:
Nit: I think this should be renamed to `ToMacFilePath` or use `Path.DirectorySeperatorChar` instead of hardcoding `'/'` just in case someone copies it somewhere.

Reply:
Good idea!